src/components/input/GeoInput/GeoModal/GeoInputList/index.js

```diff
@@ -235,17 +300,40 @@ const GeoInputList = (props) => {
             className: styles.item,
             itemKey: key,
             onDismiss: handlePolygonRemove,
-            disabled: polygonDisabled,
+            disabled: polygonDisabled || isNotDefined(onPolygonsChange),
+            readOnly,
             onEdit: handlePolygonEdit,
             value: polygon.geoJson.properties.title,
             color: polygon.geoJson.properties.color,
         }),
-        [handlePolygonEdit, handlePolygonRemove, polygonDisabled],
+        [handlePolygonEdit, handlePolygonRemove, polygonDisabled, onPolygonsChange, readOnly],
     );
 
     const newSelections = useMemo(
-        () => createCombinedSelections(selections, polygons),
-        [selections, polygons],
+        () => {
+            const tempSelections = [...createCombinedSelections(selections, polygons)]
+                .sort((a, b) => compareString(
+                    geoOptionsById[a.id].title,
+                    geoOptionsById[b.id].title,
+                ));
+            const sortOrderValue = sortOrder === 'asc' ? 1 : -1;
+
+            if (sortType === 'alphabetical') {
+                return tempSelections.sort((a, b) => compareString(
+                    geoOptionsById[a.id].title,
+                    geoOptionsById[b.id].title,
+                    sortOrderValue,
+                ));
+            } else if (sortType === 'overlaps') {
+                return tempSelections.sort((a, b) => compareNumber(
+                    (a.polygons && a.polygons.length) || -1,
+                    (b.polygons && b.polygons.length) || -1,
+                    sortOrderValue,
```

Comment:
We should have a fallback compare strategy when they have the same length of polygons.
Maybe serialize the polygon ids?
And for same polygon overlaps, have fallback for alphabetical or something

Reply:
Yeah, I sorted the data before making it the fallback.
So, if they have the same number of polygons, they'll be sorted alphabetically.

Reply:
Done